Buffer limit IOException thrown with validate-refs

### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

When I did validate-refs test:
```
validate-3.4.0-SNAPSHOT/bin/validate-refs --auth-opensearch config.properties urn:nasa:pds:orex.ovirs:data_calibrated::11.0
```
where config.properties points to https://search-en-prod-di7dor7quy7qwv3husi2wt5tde.us-west-2.es.amazonaws.com/_search

I get an IOException:
```
14:18:12.928 [main] FATAL gov.nasa.pds.validate.ri.OpensearchDocument - Error reading from URL: https://search-en-prod-di7dor7quy7qwv3husi2wt5tde.us-west-2.es.amazonaws.com/_search
java.io.IOException: entity content is too long [196292016] for the configured buffer limit [104857600]
	at org.opensearch.client.RestClient.extractAndWrapCause(RestClient.java:958) ~[opensearch-rest-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestClient.performRequest(RestClient.java:332) ~[opensearch-rest-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestClient.performRequest(RestClient.java:320) ~[opensearch-rest-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1911) ~[opensearch-rest-high-level-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1877) ~[opensearch-rest-high-level-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1845) ~[opensearch-rest-high-level-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestHighLevelClient.search(RestHighLevelClient.java:1088) ~[opensearch-rest-high-level-client-2.5.0.jar:2.5.0]
	at gov.nasa.pds.validate.ri.OpensearchDocument.search(OpensearchDocument.java:152) ~[validate-3.4.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.OpensearchDocument.load_refs(OpensearchDocument.java:114) [validate-3.4.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.OpensearchDocument.getReferencesOf(OpensearchDocument.java:189) [validate-3.4.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.Cylinder.run(Cylinder.java:47) [validate-3.4.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.Engine.processQueueUntilEmpty(Engine.java:72) [validate-3.4.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.CommandLineInterface.process(CommandLineInterface.java:121) [validate-3.4.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ReferenceIntegrityMain.main(ReferenceIntegrityMain.java:14) [validate-3.4.0-SNAPSHOT.jar:?]
Caused by: org.apache.http.ContentTooLongException: entity content is too long [196292016] for the configured buffer limit [104857600]
	at org.opensearch.client.HeapBufferedAsyncResponseConsumer.onEntityEnclosed(HeapBufferedAsyncResponseConsumer.java:92) ~[opensearch-rest-client-2.5.0.jar:2.5.0]
	at org.apache.http.nio.protocol.AbstractAsyncResponseConsumer.responseReceived(AbstractAsyncResponseConsumer.java:137) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.client.MainClientExec.responseReceived(MainClientExec.java:315) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseReceived(DefaultClientExchangeHandlerImpl.java:152) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.responseReceived(HttpAsyncRequestExecutor.java:315) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:255) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:87) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:40) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:121) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at java.lang.Thread.run(Thread.java:840) ~[?:?]
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.util.Collection.toArray()" because "c" is null
	at java.base/java.util.ArrayList.addAll(ArrayList.java:670)
	at gov.nasa.pds.validate.ri.OpensearchDocument.getReferencesOf(OpensearchDocument.java:190)
	at gov.nasa.pds.validate.ri.Cylinder.run(Cylinder.java:47)
	at gov.nasa.pds.validate.ri.Engine.processQueueUntilEmpty(Engine.java:72)
	at gov.nasa.pds.validate.ri.CommandLineInterface.process(CommandLineInterface.java:121)
	at gov.nasa.pds.validate.ReferenceIntegrityMain.main(ReferenceIntegrityMain.java:14)
```

### 🕵️ Expected behavior

I expected the tool to work

### 📜 To Reproduce

```
validate-3.4.0-SNAPSHOT/bin/validate-refs --auth-opensearch config.properties urn:nasa:pds:orex.ovirs:data_calibrated::11.0
```

### 🖥 Environment Info

Mac OSx
Java openJDK17

### 📚 Version of Software Used

Version 3.4.0-SNAPSHOT
Release Date: 2023-11-02 14:14:31

### 🩺 Test Data / Additional context

_No response_

### 🦄 Related requirements

_No response_

### ⚙️ Engineering Details

_No response_

--- a/src/main/java/gov/nasa/pds/validate/ri/OpensearchDocument.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/OpensearchDocument.java
@@ -165,7 +165,7 @@ public class OpensearchDocument implements DocumentInfo, RestClientBuilder.HttpC
   public OpensearchDocument(AuthInformation context) {
     this.context = context;
     this.hugeMemory.setHttpAsyncResponseConsumerFactory(
-        new HttpAsyncResponseConsumerFactory.HeapBufferedResponseConsumerFactory(128*1024^4)); // 128 GB
+        new HttpAsyncResponseConsumerFactory.HeapBufferedResponseConsumerFactory(128*1099511627776)); // 128 GB
   }
 
   @Override

--- a/src/main/java/gov/nasa/pds/validate/ri/OpensearchDocument.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/OpensearchDocument.java
@@ -25,6 +25,7 @@ import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.opensearch.action.search.SearchRequest;
 import org.opensearch.action.search.SearchResponse;
+import org.opensearch.client.HttpAsyncResponseConsumerFactory;
 import org.opensearch.client.RequestOptions;
 import org.opensearch.client.RestClient;
 import org.opensearch.client.RestClientBuilder;
@@ -43,6 +44,7 @@ public class OpensearchDocument implements DocumentInfo, RestClientBuilder.HttpC
   final private HashMap<String, List<String>> references = new HashMap<String, List<String>>();
   final private LidvidComparator lidvid_compare = new LidvidComparator();
   final private Logger log = LogManager.getLogger(OpensearchDocument.class);
+  final private RequestOptions.Builder hugeMemory = RequestOptions.DEFAULT.toBuilder();
 
   private void load(String lidvid) {
     if (!this.documents.containsKey(lidvid)) {
@@ -149,7 +151,7 @@ public class OpensearchDocument implements DocumentInfo, RestClientBuilder.HttpC
       try {
         if (OpensearchDocument.sourceOverride != null)
           return OpensearchDocument.sourceOverride.search(client, request);
-        return client.search(request, RequestOptions.DEFAULT);
+        return client.search(request, this.hugeMemory.build());
       } catch (ConnectException ce) {
         iteration++;
         if (iteration < 5)
@@ -162,6 +164,8 @@ public class OpensearchDocument implements DocumentInfo, RestClientBuilder.HttpC
 
   public OpensearchDocument(AuthInformation context) {
     this.context = context;
+    this.hugeMemory.setHttpAsyncResponseConsumerFactory(
+        new HttpAsyncResponseConsumerFactory.HeapBufferedResponseConsumerFactory(128*1024^4)); // 128 GB
   }
 
   @Override

--- a/src/main/java/gov/nasa/pds/validate/ri/OpensearchDocument.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/OpensearchDocument.java
@@ -165,7 +165,7 @@ public class OpensearchDocument implements DocumentInfo, RestClientBuilder.HttpC
   public OpensearchDocument(AuthInformation context) {
     this.context = context;
     this.hugeMemory.setHttpAsyncResponseConsumerFactory(
-        new HttpAsyncResponseConsumerFactory.HeapBufferedResponseConsumerFactory(128*1099511627776)); // 128 GB
+        new HttpAsyncResponseConsumerFactory.HeapBufferedResponseConsumerFactory(Integer.MAX_VALUE)); // almost 2 GB
   }
 
   @Override

--- a/src/main/java/gov/nasa/pds/validate/ri/UserInput.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/UserInput.java
@@ -9,6 +9,7 @@ import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import javax.xml.transform.sax.SAXSource;
+import org.apache.commons.lang.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.xml.sax.InputSource;
@@ -44,10 +45,11 @@ class UserInput {
           TreeInfo docInfo = LabelParser.parse(saxSource); // Parses a label.
           List<TinyNodeImpl> xmlModels = new ArrayList<>();
           XMLExtractor extractor = new XMLExtractor(docInfo.getRootNode());
-          xmlModels = extractor.getNodesFromDoc("logical_identifier");
+          xmlModels = extractor.getNodesFromDoc("//logical_identifier");
           this.labels_lidvid = xmlModels.get(0).getStringValue();
           return true;
         } catch (Exception e) {
+          e.printStackTrace();
           return false;
         }
       }
@@ -57,7 +59,10 @@ class UserInput {
   private List<String> process (List<String> cliList){
     List<String> lidvids = new ArrayList<String>();
     for (String cliArg : cliList) {
-      if (cliArg.startsWith ("urn:")) {
+      cliArg = StringUtils.substringBefore(cliArg, "#");
+      if (cliArg.length() == 0) {
+        // ignore empty lines
+      } else if (cliArg.startsWith ("urn:")) {
         lidvids.add (cliArg);
       } else if (this.isLabel (cliArg)) {
         lidvids.add (this.labels_lidvid);

